File: src/main/java/com/topface/topface/viewModels/HeaderPhotoBlogViewModel.java
package com.topface.topface.viewModels;

import android.databinding.ObservableField;
import android.text.TextUtils;

import com.topface.topface.App;
import com.topface.topface.R;
import com.topface.topface.data.Profile;
import com.topface.topface.utils.Utils;

import org.jetbrains.annotations.NotNull;

public class HeaderPhotoBlogViewModel {
    public final ObservableField<String> urlAvatar = new ObservableField<>();

    public HeaderPhotoBlogViewModel() {
        setUrlAvatar(App.get().getProfile());
    }

    public void setUrlAvatar(@NotNull Profile profile) {
        String photoUrl = profile.photo != null ? profile.photo.getDefaultLink() : Utils.EMPTY;
        urlAvatar.set(TextUtils.isEmpty(photoUrl)? Utils.getLocalResUrl(profile.sex == Profile.BOY ?
                R.drawable.upload_photo_male : R.drawable.upload_photo_female):photoUrl);
    }
}

File: src/main/java/com/topface/topface/ui/fragments/feed/photoblog/HeaderPhotoBlogViewModel.kt
package com.topface.topface.ui.fragments.feed.photoblog

import android.databinding.ObservableField
import android.text.TextUtils
import com.topface.topface.App
import com.topface.topface.R
import com.topface.topface.data.Profile
import com.topface.topface.ui.fragments.feed.feed_base.IFeedNavigator
import com.topface.topface.utils.Utils
import com.topface.topface.utils.rx.shortSubscription
import rx.Subscription

class HeaderPhotoBlogViewModel(private val mNavigator: IFeedNavigator) {
    val urlAvatar: ObservableField<String> = ObservableField(Utils.getLocalResUrl(R.drawable.upload_photo_female))
    var profileSubscription: Subscription

    init {
        setUrlAvatar(App.get().profile)
        profileSubscription = App.getAppComponent().appState().getObservable(Profile::class.java)
                .subscribe(shortSubscription { it?.let { setUrlAvatar(it) } })
    }

    fun showAddToLeader() =
            if (!App.getConfig().userConfig.isUserAvatarAvailable && App.get().profile.photo == null) {
                mNavigator.showTakePhotoPopup()
            } else {
                mNavigator.showAddToLeader()
            }

    fun setUrlAvatar(profile: Profile) {
        val photoUrl = if (profile.photo != null) profile.photo.defaultLink else Utils.EMPTY
        urlAvatar.set(if (TextUtils.isEmpty(photoUrl))
            Utils.getLocalResUrl(if (profile.sex == Profile.BOY)
                R.drawable.upload_photo_male
            else
                R.drawable.upload_photo_female)
        else
            photoUrl)
    }
}